dbms/src/Flash/Mpp/MPPTask.cpp

```diff
@@ -228,7 +228,9 @@ void MPPTask::registerTunnels(const mpp::DispatchTaskRequest & task_request)
         if (status != INITIALIZING)
             throw Exception(fmt::format("The tunnel {} can not be registered, because the task is not in initializing state", tunnel->id()));
 
-        tunnel_set_local->registerTunnel(MPPTaskId(task_meta), tunnel);
+        MPPTaskId task_id(task_meta);
+        RUNTIME_CHECK_MSG(id.gather_id.hasMeaningfulGatherId() == task_id.gather_id.hasMeaningfulGatherId(), "MPP query has gather id while mpp task does not have gather id, should be something wrong in TiDB side");
```

Comment:
why not `id.gather_id == task_id.gather_id`?

Reply:
OK